src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -0,0 +1,60 @@
+#include "abstract_table_generator.hpp"
+
+#include "benchmark_config.hpp"
+#include "benchmark_table_encoder.hpp"
+#include "operators/export_binary.hpp"
+#include "storage/storage_manager.hpp"
+
+namespace opossum {
+
+AbstractTableGenerator::AbstractTableGenerator(const std::shared_ptr<BenchmarkConfig>& benchmark_config)
+    : _benchmark_config(benchmark_config) {}
+
+void AbstractTableGenerator::generate_and_store() {
+  auto table_info_by_name = generate();
+
+  /**
+   * Encode the Tables
+   */
+  for (auto& [table_name, table_info] : table_info_by_name) {
+    table_info.re_encoded = BenchmarkTableEncoder::encode(table_name, table_info.table,
+                                                          _benchmark_config->encoding_config, _benchmark_config->out);
+  }
+
+  /**
+   * Write the Tables into binary files if required
```

Comment:
What happens if I chose an encoding that cannot be exported into binary?

Reply:
The exporter will fail with a descriptive error message and the user will hopefully not repeat his mistake.

Reply:
#1367 shouldn't be too hard to do, though